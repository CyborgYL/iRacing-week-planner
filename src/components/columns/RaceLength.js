// @flow

import React from 'react';
import { useTranslation } from 'react-i18next';

type Props = {
  race: {
    raceLength?: {
      laps: string,
      minutes: string,
    },
  },
};

export default function RaceLength({ race }: Props) {
  // console.log(race);
  const { raceLength } = race;
  const { t } = useTranslation();

  const noData = <td><div>{t('No data')}</div></td>;

  if (!raceLength) {
    return noData;
  }

  if (raceLength.laps) {
    return (
      <td><div>{t('{{laps}}L', { laps: raceLength.laps })}</div></td>
    );
  }

  if (raceLength.minutes) {
    if (raceLength.minutes > 0 && raceLength.minutes % 60 === 0) {
      return (
        <td><div>{t('{{hours}}h', { hours: raceLength.minutes / 60 })}</div></td>
      );
    }
    return (
      <td><div>{t('{{minutes}}m', { minutes: raceLength.minutes })}</div></td>
    );
  }

  return noData;
}
