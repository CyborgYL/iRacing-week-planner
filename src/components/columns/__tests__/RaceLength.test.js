import React from 'react';
import renderer from 'react-test-renderer';
import { describe, test } from '@jest/globals';

import RaceLength from '../RaceLength';

describe('components/columns/RaceLength', () => {
  test('renders null', () => {
    const component = renderer.create(
      <RaceLength
        race={{
          raceLength: null
        }}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders laps', () => {
    const component = renderer.create(
      <RaceLength
        race={{
          raceLength: {
            laps: 45,
          }
        }}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders hours', () => {
    const component = renderer.create(
      <RaceLength
        race={{
          raceLength: {
            minutes: 120,
          }
        }}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders minutes', () => {
    const component = renderer.create(
      <RaceLength
        race={{
          raceLength: {
            minutes: 45,
          }
        }}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
