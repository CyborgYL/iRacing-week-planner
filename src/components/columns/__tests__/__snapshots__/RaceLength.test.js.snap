// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/RaceLength renders empty object mistake 1`] = `
<td>
  <div>
    No data
  </div>
</td>
`;

exports[`components/columns/RaceLength renders hours 1`] = `
<td>
  <div>
    2h
  </div>
</td>
`;

exports[`components/columns/RaceLength renders laps 1`] = `
<td>
  <div>
    45L
  </div>
</td>
`;

exports[`components/columns/RaceLength renders minutes 1`] = `
<td>
  <div>
    45m
  </div>
</td>
`;

exports[`components/columns/RaceLength renders null 1`] = `
<td>
  <div>
    No data
  </div>
</td>
`;
